### PR TITLE
Add alternate buttons for scrolling through inventory

### PR DESCRIPTION
**Pull request by brickman1444**
for people without scroll wheels, like Zac

### CODE REVIEW

File: Assets/Scripts/Player/PlayerInventory.cs [hunk 1]
```diff
@@ -58,19 +60,39 @@ void Start()
...
-			int nextIndex = resourceIndex + Mathf.Clamp( Mathf.RoundToInt( scrollAmount ), -1, 1 );
-			resourceIndex = MathUtils.Mod( nextIndex, heldResources.Count );
-			SpawnResourceObject();
+			// Nothing to scroll
+			return;
+		}
+
+		float scrollAmount = InputUtils.GetAxis("Scroll");
+
+		if ( scrollAmount < WadeUtils.SMALLNUMBER && scrollAmount > -WadeUtils.SMALLNUMBER )
```
Comment (randomPoison) [inline Assets/Scripts/Player/PlayerInventory.cs]: I think there's an `IsZero()` method for floats in `MathUtils`. If not, now would be a good time to add one.

File: Assets/Scripts/Player/PlayerInventory.cs [hunk 2]
```diff
@@ -58,19 +60,39 @@ void Start()
...
+			{
+				scrollAmount = 1f;
+			}
+			else if ( _altScrollButton.negativeDown )
+			{
+				scrollAmount = -1f;
+			}
+			else
+			{
+				return;
```
Comment (randomPoison) [inline Assets/Scripts/Player/PlayerInventory.cs]: The control flow here isn't very clear, it took me a long time to understand what's going on here. Get rid of the early return, it's not necessary and only makes the method harder to read.

**Comment by randomPoison** [comment]
I'm done looking this one over.

File: Assets/Scripts/Player/PlayerInventory.cs [hunk 3]
```diff
@@ -58,16 +60,35 @@ void Start()
...
+				scrollAmount = -1f;
+			}
+		}
+
+		if ( !WadeUtils.IsZero( scrollAmount ) )
+		{
+			int nextIndex = resourceIndex + ( scrollAmount < 0f ? -1 : 1 );
 			resourceIndex = MathUtils.Mod( nextIndex, heldResources.Count );
 			SpawnResourceObject();
 		}
```
Comment (brickman1444) [inline Assets/Scripts/Player/PlayerInventory.cs:94]: How is this for readability?

Comment (excaliburHisSheath) [inline Assets/Scripts/Player/PlayerInventory.cs]: Use an `else` statement so you're not doing two conditional branches.

Comment (randomPoison) [inline Assets/Scripts/Player/PlayerInventory.cs:94]: Add a comment on that second block to explain why you need to check for zero a second time.

**Pull request merged**